.ci/jenkins/ci.groovy

```diff
@@ -95,18 +108,34 @@ CONFIGURATION_WORKFLOW = { configuration ->
     }
 }
 
-def cloneRepository(String address, String branch) {
+def gitClone(String label, String address, String branch) {
     repositoryName = address.split("/").last().replace(".git","")
-    dir ("${WORKDIR}/${repositoryName}") {
-        retry(3) {
-            checkout([$class: 'GitSCM',
-                branches: [[name: "${branch}"]],
-                doGenerateSubmoduleConfigurations: false,
-                extensions: [[$class: 'CloneOption', depth: 1, noTags: false, shallow: true, timeout: 30]],
-                submoduleCfg: [],
-                userRemoteConfigs: [[credentialsId: "${JENKINS_GITHUB_CREDENTIAL_ID}",
-                url: "${address}"]]])
-        }
+
+    sh  label: label,
+        script:
+    """
+        git clone \
+            -b ${branch} \
+            --single-branch \
+            --no-tags \
+            --dissociate \
+            --depth 1 \
+            --verbose \
+            ${address} \
+            ${WORKDIR}/${repositoryName}
+    """
+}
+
+def gitSubmoduleUpdate(String repository_name, String exception="") {
```

Comment:
What is parameter "String exception" for?

Reply:
You are right, it is a hangover of excluding to update some submodule of a particular repository. But for now it is useless (not used in definition code), I will fix it.